geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/Runner.scala

```diff
@@ -79,6 +79,57 @@ trait Runner {
     }
   }
 
+
+  def autocompleteUsage(jc: JCommander): String = {
+    val commands = jc.getCommands.map(_._1).toSeq
+    val out = new StringBuilder
+    out.append(
+      s"""{ # This function is named by the script generating this so it matches the caller script's name.
+         |  local cur prev;
+         |  COMPREPLY=();
+         |  cur="$${COMP_WORDS[COMP_CWORD]}";
+         |  prev="$${COMP_WORDS[COMP_CWORD-1]}";
+         |
+         |  if [[ "$${COMP_WORDS[1]}" == "help" ]]; then
+         |    COMPREPLY=( $$(compgen -W "${commands.mkString(" ")}" $${cur}));
+         |    return 0;
+         |  fi;
+         |
+         |  case $${COMP_CWORD} in
+         |    1)
+         |      COMPREPLY=( $$(compgen -W "${commands.mkString(" ")}" $${cur}));
+         |      ;;
+         |    [2-9] | [1-9][0-9])
+         |      if [[ "$${cur}" =~ ^-[a-zA-Z-]?+$$ ]]; then
+         |        case $${COMP_WORDS[1]} in
+        """.stripMargin)
+    commands.foreach { command =>
+      val params = jc.getCommands.get(command).getParameters.filter(!_.getParameter.hidden()).flatMap(_.getParameter.names().filter(_.length != 2))
+        out.append(
+      s"""            $command)
+         |              COMPREPLY=( $$(compgen -W "${params.mkString(" ").replace(",", " ").replace("  "," ")}" -- $${cur}));
```

Comment:
might want to use a regex instead of `.replace(",", " ").replace("  "," ")` - maybe this won't happen given the inputs, but if there were more than 2 sequential spaces they might not get replaced correctly. Instead you could do:
`.replaceAll("[,\\s]+", " ")`

```
scala> "1,  2".replace(",", " ").replace("  "," ")
res0: String = 1  2
scala> "1,  2".replaceAll("[,\\s]+", " ")
res1: String = 1 2
```